text/0000-logger_support.md

```diff
@@ -0,0 +1,68 @@
+- Feature Name: `Logger support`
+- Start Date: 2019-12-28
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add support for interrupt priority based logging in RTFM (ala [`cortex-m-funnel`]).
+
+# Motivation
+[motivation]: #motivation
+
+One of the most time consuming issues of embedded programming is debugging. And especially when it comes to event based systems where debugging can get in the way of timing requirements. Debugging is often done through logging with `printf`-like methods, and the logger implementation will often impede real-time requirements, causing the system to behave differently with and without logging enabled.
+This often comes from logger implementations having global critical sections for accessing the log buffers, but with RTFM we can have seamless support for logging without critical sections by leveraging the design and ideas behind [`cortex-m-funnel`].
+
+The idea behind it is that there exists one logging queue per interrupt priority, and a logging function always writes to the currently running priority which in turn means that no critical sections are needed (see [`cortex-m-funnel`] for implementation details).
+
+Adding this feature will help users immensely, as figuring out logging is not a trivial task. This will also help adoption of RTFM as **we made logging easy**.
+
+# Detailed design
+[design]: #detailed-design
+
+The important thing is to define how large the buffers should be. Either we can go with the simplest case (and have all buffers the same size):
+
+```rust
+#[app(logger = 256)]
+// ...
+```
+
+Or we can have a base size, together with possible specialization (bikeshed of API needed):
+
+```rust
+// Base size of 256 bytes, but the queue for priority 7 is of 1024 bytes
+#[app(logger = [base = 256, 7 = 1024])]
```

Comment:
The syntax of the `#[app]` arguments can be anything; the downside is that it never gets formatted by `rustfmt`. I personally would prefer not to enter so much data as an attribute arguments but can't think of a better syntax right now.